src/modules/c.rs

```diff
@@ -0,0 +1,216 @@
+    let is_c_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .set_files(&config.detect_files)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_c_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        let c_compiler_info = Lazy::new(|| {
+            context.exec_cmds_return_first(&[
+                // the compiler is usually cc, and --version works on gcc and clang
+                &["cc", "--version"],
+                // but on some platforms gcc is installed as *gcc*, not cc
+                &["gcc", "--version"],
+                // for completeness, although I've never seen a clang that wasn't cc
+                &["clang", "--version"],
+            ])
+        });
+
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "compiler_name" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+
+                    let c_compiler = if c_compiler_info.contains("clang") {
+                        "clang"
+                    } else if c_compiler_info.contains("Free Software Foundation") {
+                        "gcc"
+                    } else {
+                        return None;
+                    };
+                    Some(c_compiler).map(Cow::Borrowed).map(Ok)
+                }
+                "compiler_version" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+                    if c_compiler_info.contains("clang")
+                        || c_compiler_info.contains("Free Software Foundation")
+                    {
+                        // Clang says ...
+                        //   Apple clang version 13.0.0 ...\n
+                        //   OpenBSD clang version 11.1.0\n...
+                        //   FreeBSD clang version 11.0.1 ...\n
+                        // so seems to have the version always in position 3.
+                        // gcc says ...
+                        //   gcc (OmniOS 151036/9.3.0-il-1) 9.3.0\n...
+                        //   gcc (Debian 10.2.1-6) 10.2.1 ...\n
+                        // so again in position 3.
+                        // But this is CRUFTY AS HELL
+                        VersionFormatter::format_module_version(
+                            module.get_name(),
+                            c_compiler_info.split_whitespace().nth(3)?,
+                            config.version_format,
+                        )
+                        .map(Cow::Owned)
+                        .map(Ok)
+                    } else {
```

Comment:
Nah. I'm happy with just not doing anything if we don't recognise the compiler. Any user who cares will notice pretty darned quickly that their compiler isn't recognised and the solution, whether they notice because it's not in the prompt or because they notice an error message, is the same - raise a PR (or an issue). I'll put a note in the docs about this, and I'd be happy to deal with such issues.